From d2fca615935e3003cd597d4a71d81522976abf6f Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Fri, 23 Feb 2018 12:39:22 +0200
Subject: [PATCH 3375/5242] MLK-17459-4: drm: imx: dcss: fix weston

commit  2f4a8377a33a43f9a40aec8a438c64a3fca3082d from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch fixes an issue introduced by the cropping patches which made
weston look bad. That's because use_dtrc flag was enabled if modifiers
were present. However, graphics plane can have modifiers too. This patch
adds an extra check.

Also, remove an unnecessary debug message.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-dpr.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/gpu/imx/dcss/dcss-dpr.c b/drivers/gpu/imx/dcss/dcss-dpr.c
index 8bc2791a..15f1522 100644
--- a/drivers/gpu/imx/dcss/dcss-dpr.c
+++ b/drivers/gpu/imx/dcss/dcss-dpr.c
@@ -616,13 +616,11 @@ void dcss_dpr_format_set(struct dcss_soc *dcss, int ch_num, u32 pix_format,
 	struct drm_format_name_buf format_name;
 	enum dcss_color_space dcss_cs;
 
-	dev_dbg(dcss->dev, "%s\n", __func__);
-
 	dcss_cs = dcss_drm_fourcc_to_colorspace(pix_format);
 	ch->planes = drm_format_num_planes(pix_format);
 	ch->bpp = dcss_dpr_get_bpp(pix_format);
 	ch->pix_format = pix_format;
-	ch->use_dtrc = modifiers_present;
+	ch->use_dtrc = ch_num && modifiers_present;
 
 	dev_dbg(dcss->dev, "pix_format = %s, colorspace = %d, bpp = %d\n",
 		drm_get_format_name(pix_format, &format_name), dcss_cs, ch->bpp);
-- 
1.7.9.5

